### PR TITLE
gox: use buildGoModule

### PR DESCRIPTION
**Pull request by azahi**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by azahi** [comment]
I can't make wrapper creation to work here for some reason:
```
gox> post-installation fixup
gox> shrinking RPATHs of ELF executables and libraries in /nix/store/kbnh1mafh1bf0gnybs6rhgbyx32nzvda-gox-1.0.1
gox> shrinking /nix/store/kbnh1mafh1bf0gnybs6rhgbyx32nzvda-gox-1.0.1/bin/gox
gox> patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
gox> strip is /nix/store/61zfi5pmhb0d91422f186x26v7b52y5k-gcc-wrapper-11.3.0/bin/strip
gox> stripping (with command strip and flags -S) in /nix/store/kbnh1mafh1bf0gnybs6rhgbyx32nzvda-gox-1.0.1/bin
gox> patching script interpreter paths in /nix/store/kbnh1mafh1bf0gnybs6rhgbyx32nzvda-gox-1.0.1
gox> checking for references to /build/ in /nix/store/kbnh1mafh1bf0gnybs6rhgbyx32nzvda-gox-1.0.1...
gox> patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
error: output '/nix/store/kbnh1mafh1bf0gnybs6rhgbyx32nzvda-gox-1.0.1' is not allowed to refer to the following paths:
         /nix/store/4mz4navg6bwavsfkwzpphbsm284ns087-go-1.17.10
```
Any help would be appreciated.

File: pkgs/development/tools/gox/default.nix [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
     repo = "gox";
     rev = "v${version}";
     sha256 = "0mkh81hd7kn45dz7b6yhzqsg2mvg1g6pwx89jjigxrnqhyg9vrl7";
   };
+  vendorSha256 = null;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postFixup = ''
+    wrapProgram $out/bin/gox --prefix PATH : ${lib.makeBinPath [ go ]}
```
Comment (azahi) [inline pkgs/development/tools/gox/default.nix:23]: This does not work :(

Comment (trofi) [reply pkgs/development/tools/gox/default.nix:23]: Try allowing `go` reference explicitly (AFAIU that's your intention):

```diff
--- a/pkgs/development/tools/gox/default.nix
+++ b/pkgs/development/tools/gox/default.nix
@@ -19,6 +19,8 @@ buildGoModule rec {

   nativeBuildInputs = [ makeWrapper ];

+  # We are about to embed path to `go` bianry:
+  allowGoReference = true;
   postFixup = ''
     wrapProgram $out/bin/gox --prefix PATH : ${lib.makeBinPath [ go ]}
   '';
```

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 176478` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gox</li>
  </ul>
</details>

**Pull request merged**